test/integration/hook-err.spec.js

```diff
@@ -39,7 +39,7 @@ describe('hook error handling', function() {
   describe('after each hook error', function() {
     before(run('hooks/afterEach-hook-error.fixture.js'));
     it('should verify results', function() {
-      assert.deepEqual(lines, ['test 1', 'after', bang + 'test 3']);
+      assert.deepEqual(lines, ['test 1', 'after', bang, 'test 3']);
```

Comment:
is this related?

Reply:
https://github.com/mochajs/mocha/blob/8368722830ed80e03a52d29155310cdc7b6fc123/test/integration/hook-err.spec.js#L159 
In here,`splitRegExp` is `[\n.]+`.
So, prev result is `.!test 3`, but now the result is `!.test 3`.
Therefore, `!` and `test 3` is separated.